### PR TITLE
Add the releaser workflows to the repo

### PR DESCRIPTION
**Pull request by jtpio**
## References

Fixes https://github.com/jupyterlab/team-compass/issues/165

## Code changes

- [x] Add the Jupyter Releaser workflows to the repo so releases can more easily be made from https://github.com/jupyterlab/jupyterlab/actions
- [x] Skip `ensure-sha` for now https://github.com/jupyterlab/jupyterlab/issues/13442
- [x] Add `jupyterlab-bot` to the repo: https://github.com/jupyterlab-bot
- [x] Add `jupyterlab-releaser` to PyPI: https://pypi.org/user/jupyterlab-releaser
- [x] Add `jupyterlab-releaser-bot` to `npm`: https://www.npmjs.com/~jupyterlab-release-bot
- [x] Update `RELEASE.md` (update instructions and workflow names to releaser v2)
- [x] Add `ADMIN_TOKEN`
- [x] Add `PYPI_TOKEN`
- [x] Add `NPM_TOKEN`

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      branch:
+        description: "The target branch"
+        required: false
+      release_url:
+        description: "The URL of the draft GitHub release"
+        required: false
+      steps_to_skip:
+        description: "Comma separated list of steps to skip"
+        required: false
+        default: "ensure-sha"
```
Comment (jtpio) [inline .github/workflows/publish-release.yml:14]: Keeping `ensure-sha` as default here for now as I believe https://github.com/jupyterlab/jupyterlab/issues/13442 is still an issue with Releaser v2 (cc @blink1073 for confirmation).

**Comment by blink1073** [comment]
I updated the tokens

File: .github/workflows/publish-release.yml (hunk 1 above)
Comment (blink1073) [reply .github/workflows/publish-release.yml:14]: Yes, that is correct

**Review by blink1073 — APPROVED** [review]
LGTM!

**Comment by jtpio** [comment]
Thanks!

**Comment by jtpio** [comment]
> I updated the tokens

@blink1073 curious which token has been set for `ADMIN_GITHUB_TOKEN`? Is it a fine grained token with an expiration time?

I tried starting a new release from the repo but kept getting `401` errors, even after adding myself as "Admin" to the repo directly. Could this be because the bot token is expired? https://github.com/jupyterlab/jupyterlab/actions/workflows/prep-release.yml

Starting the workflow from the fork seems to go past the auth step: https://github.com/jtpio/jupyter_releaser/actions/runs/4794420430/jobs/8527782598

**Comment by jtpio** [comment]
Ah ok thanks for the extra context.

Then the 90 days expiration is unlikely to be the issue since it's been less than 3 months. Maybe it could be something related to how the releaser handles users that are part of the multiple teams which have different rights? Although that too used to be fine before iirc.

**Comment by blink1073** [comment]
` RuntimeError: Could not get user permission level, assuming user was not admin!` means that the admin token itself did not work.  I created the "release token apr 2023" token 11 days ago, my guess is JupyterLab is using the expired "release token jan 2023" token.

**Comment by jtpio** [comment]
Ah ok, it says it was updated March 13 here:

image

But it could indeed be an older one.

Would it be possible to update the token? I don't have access to the bot account.

Also what could be a way for handling expired tokens in the future?

**Comment by jtpio** [comment]
ok thanks!

I'll open a new issue in the team compass to track this since this is something that might affect other repos in the `jupyterlab` org using the releaser bot.

**Comment by jtpio** [comment]
Opened https://github.com/jupyterlab/team-compass/issues/190

**Pull request merged**